data/transactions/logic/eval.go

```diff
@@ -1254,6 +1255,22 @@ func opB(cx *evalContext) {
 	cx.nextpc = target
 }
 
+func opCallSub(cx *evalContext) {
+	cx.callstack = append(cx.callstack, cx.pc+3)
```

Comment:
I suggest to use a hidden stack for a chain of ret addresses with call to store and ret to pop. No other access. This is because of a long history of security problems in x86 and ret addresses on a stack.

Reply:
Yes, that's what this is.  cx.callstack inaccessible from any other instruction, except retsub accesses and returns to.